mesonbuild/options.py

```diff
@@ -812,10 +812,11 @@ def __init__(self, is_cross: bool) -> None:
         self.augments: OptionDict = {}
         self.is_cross = is_cross
 
-        # Pending options are options that need to be initialized later, either
-        # configuration dependent options like compiler options, or options for
-        # a different subproject
+        # Pending options are configuration dependent options that could be
+        # initialized later, such as compiler options
         self.pending_options: OptionDict = {}
+        # Subproject options from toplevel project()
+        self.pending_subproject_options: OptionDict = {}
 
     def clear_pending(self) -> None:
```

Comment:
Should this method not also clear `pending_subproject_options`?

Reply:
No, subproject options could come back to life in a future invocation if a change in project options triggers the processing of a previously-unused subproject.

Actually I don't think `clear_pending` should exist at all: either an option is unknown, or it should remain available for future use. I didn't want to include this behavioral change in the series, but I can try to add it on top.

Reply:
But it also means that if the user edits their build file and _then_ triggers a previously unused subproject, the option values are the ones from when the build directory was first set up, not what they are now.

Reply:
Isn't that #2193? clear_pending doesn't fix it, it just makes the behavior different depending on whether the option was used before